### PR TITLE
Fix SNAT pool allocation for APIC ML2

### PR DESCRIPTION
**Pull request by tbachman**
This fixes a bug where the wrong SNAT subnet could
be picked when used in APIC ML2 mode.

It also adds the SNAT subnet to the "extra_subnets"
attribute passed in the gateway port to the config agent.

This closes issue #240

Signed-off-by: Thomas Bachman tbachman@yahoo.com

### CODE REVIEW

**Comment by tbachman** [comment]
Pushed update to patch (bugs fixed after functional testing).

File: networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py [hunk 1]
```diff
@@ -218,15 +219,45 @@ def apic_driver(self):
...
+        if subnet['network_id'] == net['id']:
+            return True
+
+        network = self._core_plugin.get_network(
+            context.elevated(), subnet['network_id'])
+        ext_net_name = network['name']
+        if (APIC_SNAT_NET + '-') in ext_net_name:
+            # This is APIC ML2 mode -- we need to strip the prefix
+            ext_net_name = ext_net_name.strip(APIC_SNAT_NET + '-')
+        if net['id'] == ext_net_name:
```
Comment (kentwu1) [inline networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py]: I think this if-else block should be moved under the if block at line 240.

Comment (tbachman) [inline networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py]: I think I see what you're saying. Will fix.

**Comment by tbachman** [comment]
Pushed new version, addressing reviewer comments (tested GBP and APIC ML2).

**Pull request merged**